app/ui-message/client/messageBubble.js

```diff
@@ -0,0 +1,315 @@
+import { Tracker } from 'meteor/tracker';
+import { Template } from 'meteor/templating';
+import { TAPi18n } from 'meteor/rocketchat:tap-i18n';
+
+import { MessageTypes } from '../../ui-utils/client';
+import { Markdown } from '../../markdown/client';
+import './messageThread';
+import { escapeHTML } from '../../../lib/escapeHTML';
+import { renderMentions } from '../../mentions/client/client';
+import { renderMessageBody } from '../../../client/lib/renderMessageBody';
+import { settings } from '../../settings/client';
+import './messageBubble.html';
+
+const renderBody = (msg, settings) => {
+	const searchedText = msg.searchedText ? msg.searchedText : '';
+	const isSystemMessage = MessageTypes.isSystemMessage(msg);
+	const messageType = MessageTypes.getType(msg) || {};
+
+	if (messageType.render) {
+		msg = messageType.render(msg);
+	} else if (messageType.template) {
+		// render template
+	} else if (messageType.message) {
+		msg.msg = escapeHTML(msg.msg);
+		msg = TAPi18n.__(messageType.message, { ...typeof messageType.data === 'function' && messageType.data(msg) });
+	} else if (msg.u && msg.u.username === settings.Chatops_Username) {
+		msg.html = msg.msg;
+		msg = renderMentions(msg);
+		msg = msg.html;
+	} else {
+		msg = renderMessageBody(msg);
+	}
+
+	if (isSystemMessage) {
+		msg.html = Markdown.parse(msg.html);
+	}
+
+	if (searchedText) {
+		msg = msg.replace(new RegExp(searchedText, 'gi'), (str) => `<mark>${ str }</mark>`);
+	}
+
+	return msg;
+};
+
+Template.messageBubble.helpers({
+	unread() {
+		const { msg, subscription } = this;
+		return subscription?.tunread?.includes(msg._id);
+	},
+	body() {
+		const { msg, settings } = this;
+		return Tracker.nonreactive(() => renderBody(msg, settings));
+	},
+	attachmentsClass() {
+		const { msg } = this;
+
+		if (msg?.attachments?.length) {
+			if (msg.attachments[0]?.image_type) {
+				return 'attachment image';
+			}
+
+			return 'attachment file';
+		}
+
+		if (msg.urls?.length) {
+			return 'url';
+		}
+	},
+	hasAttachments() {
+		const { msg } = this;
+		return msg.attachments?.length;
+	},
+	getFileIcon() {
+		const { msg } = this;
+
+		if (msg?.attachments?.length) {
+			if (msg.attachments[0]?.image_type) {
+				return false;
+			}
+			if (msg.attachments[0]?.audio_type) {
+				return false;
+			}
+			if (msg.attachments[0]?.video_type) {
+				return false;
+			}
+
+			let ext = msg.attachments[0]?.title_link;
+			ext = ext.substring(ext.lastIndexOf('.'));
+
+			if (ext === '.doc') {
+				return 'file-document';
+			}
+			if (ext === '.pdf') {
+				return 'file-pdf';
+			}
+			if (ext === '.key') {
+				return 'file-keynote';
+			}
+			if (ext === '.zip') {
+				return 'files-zip';
+			}
+
+			return 'file-generic';
+		}
+	},
+	own() {
+		const { msg, u = {} } = this;
+		return msg.u && msg.u._id === u._id ? 'own' : 'notOwn';
```

Comment:
Can we change the classname to make it more clear here

Reply:
Done, Thanks